src/components/Timeline/Timeline.js

```diff
@@ -29,6 +29,14 @@ class Timeline extends Component {
     }
   }
 
+  componentDidUpdate (oldProps) {
+    const { dispatch, incidentId, ticketId } = this.props
+    if (oldProps.incidentId !== incidentId) {
+      dispatch(eventActions.fetchEvents({ ticketId: ticketId, incidentId: incidentId }))
```

Comment:
I don't think this preserves the event type and other filters.

I think this will put us in a situation where the filter component may show that only a few event types are selected, but the API calls will load all event types.

SynchronizeFilters includes a call to fetchEvents; it may be best to use that.

Reply:
Replacing `dispatch(eventActions.fetchEvents({ ticketId: ticketId, incidentId: incidentId }))` with `dispatch(filterActions.synchronizeFilters(filter, incidentId, ticketId, history))` results in buggy behavior. Specifically, using just synchronizeFilters causes the event timeline to be depopulated when a user clicks on a link. If I keep both lines, the behavior is correct, but I'm confused, because from what you've told me, synchronizeFilters should be implicitly fetching.

Reply:
We don't do client side filtering on anything but incidentId because our pagination library only supports filtering on one thing at a time; when we change filter we drop the event collection and reload it from the server. I think the bug may come from filteractions.synchronizeFilters:

    const newFilter = Object.assign({ incidentId: incidentId, ticketId: ticketId }, filter)

If the passed in filter has a different incidentId from the incidentId passed into the function, it will use the passed in (*old*) filter. That causes it to load events for the old incidentId, but UpdatePagination will cause our pagination library to filter down to only events with the new incidentId, leaving none. Since you're passing the new filter into eventActions.fetchEvents, that gets the events for the new incident, which may be why the behavior is right when you have both.

    const newFilter = Object.assign({}, filter, { incidentId: incidentId, ticketId: ticketId })

^This in synchronizeFilters may result in the correct behavior using sychronizeFilters alone.

Reply:
s̵͎̝̜̏̿͝u̸̪̜̞̾́̊ḇ̵̪͓̓͒̒t̷̨͓̖̀̃͂l̷̢̲̥̆̈́͠e̸̫͇̫͊̋̆ ̴̼̯̼̆̊̚b̴̢̼̳͊̑́u̸̢͎͙͐͋̽ģ̸̱̭̊̀͝s̵͔̮̝͗̓̄

Reply:
@pdimitratos Theory confirmed: <img width="668" alt="screen shot 2018-02-21 at 3 59 43 pm" src="https://user-images.githubusercontent.com/509611/36512697-6f528d58-1720-11e8-95e7-ff4f5aeca1c5.png">

Reply:
I especially like the `Object.assign({},` syntax, though it is too clever by half, it does great coercing for empty values.